lib/name_drop/client.rb

```diff
@@ -66,23 +66,27 @@ def post(endpoint, attributes)
     # @param [Hash] attributes
     # @return [Hash] encapsulates Mention API response of PUT request
     def put(endpoint, attributes)
-      request = Net::HTTP::Put.new request_uri(endpoint)
-      request.body = JSON.dump(attributes)
-      execute_request request
+      make_request(endpoint, 'Put', attributes)
     end
 
     # Makes DELETE Request through RestClient::Request.execute
     #
     # @param [String] endpoint
     # @return [Hash] encapsulates Mention API response of DELETE request
     def delete(endpoint)
-      execute_request Net::HTTP::Delete.new(request_uri(endpoint))
+      make_request(endpoint, 'Delete')
     end
 
     # @!endgroup
 
     private
 
+    def make_request(endpoint, method, attributes = {})
+      request = "Net::HTTP::#{method}".constantize.new request_uri(endpoint)
+      request.body = JSON.dump(attributes) unless attributes.empty?
+      execute_request request
```

Comment:
I kinda like the functional composure style. What do you think about this just returning the request and the callers (`delete`, `put`, etc) pass it to execute
```ruby
execute_request build_request(endpoint, 'Put', attributes)`)
```